server/src/main/java/org/elasticsearch/action/datastreams/lifecycle/ExplainIndexDataStreamLifecycle.java

```diff
@@ -79,7 +79,7 @@ public ExplainIndexDataStreamLifecycle(StreamInput in) throws IOException {
             this.rolloverDate = in.readOptionalLong();
             this.generationDateMillis = in.readOptionalLong();
             this.lifecycle = in.readOptionalWriteable(DataStreamLifecycle::new);
-            this.error = in.readOptionalString();
+            this.error = in.readOptionalWriteable(ErrorEntry::new);
```

Comment:
No BWC guards needed as 8.11 is not released yet.

Reply:
Done in https://github.com/elastic/elasticsearch/pull/101475/commits/55e19146c26f83d4e5c755118bb300c9e910ae91